driver-core/src/main/java/com/datastax/driver/core/ErrorTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+/**
+ * Interface for objects that are interested in monitoring precisely errors returned by all hosts.
+ *
+ * An implementation of this interface can be registered against a Cluster
+ * object trough the {@link Cluster#register} method, after which the
+ * {@link #notifyError(Host, Statement, Exception)} method will be called after each query of the driver to a Cassandra
+ * host with the error that has been returned for this operation.
+ */
+public interface ErrorTracker {
+
+    /**
+     * Method that is called after each request returning an error code from Cassandra.
+     *
+     * @param host the host that has been queried.
+     * @param statement the statement that has been sent.
+     * @param exception the exception the query caused.
+     */
+    public void notifyError(Host host, Statement statement, Exception exception);
```

Comment:
Not sold on `ErrorTracker` either, I agree that `LatencyTracker` should be rebranded to something more generic, e.g. `ExecutionTracker`, but that would be a breaking change.